src/query/service/src/interpreters/hook/vacuum_hook.rs

```diff
@@ -65,3 +68,16 @@ pub fn hook_vacuum_temp_files(query_ctx: &Arc<QueryContext>) -> Result<()> {
 
     Ok(())
 }
+
+pub fn hook_disk_temp_dir(query_ctx: &Arc<QueryContext>) -> Result<()> {
+    let mgr = TempDirManager::instance();
+
+    if mgr.drop_disk_spill_dir(&query_ctx.get_id())? && rand::thread_rng().gen_ratio(1, 10) {
+        let deleted = mgr.drop_disk_spill_dir_unknown(5)?;
```

Comment:
Why not vacuum all temporary files here.

Reply:
conservative strategy

Reply:
We can vacuum all temp files here by default and add a setting to limit it when necessary.